### PR TITLE
Release Changes for master

### PR DESCRIPTION
**Pull request by felixboehm**
- better Makefile
- added .gitignore

Open tasks for using default Makefile
- [ ] add webpack build to default Makefile 
- [ ] add testing to default Makefile
- [ ] update README.md on how to use Makefile
- [ ] npm dependencies cleanup?
  - vue -> production dependency
  - rest -> dev dependency

### CODE REVIEW

File: appinfo/info.xml [hunk 1]
```diff
@@ -22,7 +22,7 @@ Furthermore guests are fully auditable with the [ownCloud Auditing application](
 	<documentation>
     		<user>https://doc.owncloud.com/server/10.0/user_manual/files/webgui/sharing.html?highlight=guest#sharing-files-with-guest-users</user>
     	</documentation>
-	<licence>AGPL</licence>
+	<licence>GPLv2.0</licence>
```
Comment (DeepDiver1975) [inline appinfo/info.xml]: needs to be added to marketplace first

**Comment by PVince81** [comment]
Better Makefile but not as good as the other ones (this one seems to be different). Running `make` twice will rerun npm install every time due to missing a "package.json" related rule.

Doesn't matter, as long as `make dist` works it's fine for now.

See comment from @DeepDiver1975 before merging.

File: appinfo/info.xml (hunk 1 above)
Comment (felixboehm) [reply appinfo/info.xml]: is wip

**Comment by felixboehm** [comment]
Please keep release branch, until we decide how to persist release version.

File: appinfo/info.xml (hunk 1 above)
Comment (IljaN) [reply appinfo/info.xml]: @felixboehm can you please use GPLv2 as license key to be in line with other license types? GPLv2 is also the official abbreviation accorind to https://www.gnu.org/licenses/gpl-2.0.html

**Comment by IljaN** [comment]
If we want to release this under GPLv2 we should change the license headers in all of the source files

**Comment by DeepDiver1975** [comment]
> If we want to release this under GPLv2 we should change the license headers in all of the source files

@IljaN done

**Pull request merged**